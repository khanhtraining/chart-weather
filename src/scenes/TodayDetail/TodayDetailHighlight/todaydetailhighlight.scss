.todayDetailHighlight-wrapper {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 40%;
    padding: 1%;
    .todayDetailHighlight-wrapper__icon {
        width: 80px;
        height: 80px;
        background-size: contain;
        cursor: pointer;
    }

    .todayDetailHighlight-wrapper__info {
        margin-left: 30px;
        font-size: 20px;

        .highlight-text {
            font-size: 30px;
            color: #ffffff;
        }
        .highlight-info {
            .highlight-temperature:before {
                display: inline-block;
                width: 20px;
                height: 20px;
                background-size: contain;
                background-image: url("../../../assets/temperature.svg");
                content: "";
                color: #ffffff;
            }
            .highlight-humidity::before {
                margin-left: 40px;
                display: inline-block;
                width: 20px;
                height: 20px;
                background-size: contain;
                background-image: url("../../../assets/humidity.svg");
                content: "";
                color: #ffffff;
            }
        }
    }
}
.location {
    text-align: center;
}
