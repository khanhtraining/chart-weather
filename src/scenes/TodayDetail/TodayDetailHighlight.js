import React from 'react'
import PropTypes from 'prop-types'
import './todaydetailhighlight.scss'
const TodayDetailHighlight = (props) => {
    return (
        <React.Fragment>
            <div className="todayDetailHighlight-wrapper">
                <div className="todayDetailHighlight-wrapper__icon">
                    {
                        props.todayHighLight ? <img src={`http://openweathermap.org/img/wn/${props.todayHighLight?.icon}@2x.png`} alt='cloud-icon' /> : <>
                        </>
                    }
                </div>
                <div className="todayDetailHighlight-wrapper__info">
                    {props.todayHighLight ? <div className="highlight-text">{props.todayHighLight?.main}</div> : <>
                    </>}
                    <div className="highlight-info">
                        <span className="highlight-temperature">{Math.round((props.todayDeTail?.main?.temp) / 10)}Â°C</span>
                        <span className="highlight-humidity">{Math.round((props.todayDeTail?.main?.humidity))}%</span>
                    </div>
                </div>
            </div>
        </React.Fragment>
    )
}

TodayDetailHighlight.propTypes = {
    locationId: PropTypes.string
}

export default TodayDetailHighlight
