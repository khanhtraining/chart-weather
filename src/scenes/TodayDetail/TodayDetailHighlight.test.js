import React from 'react'
import renderer from 'react-test-renderer'
import TodayDetailHighlight from './TodayDetailHighlight'

describe('snapshot testing', () => {
    test('snapshot for today detail highlight component', () => {
        const renderedComponent = renderer.create(<TodayDetailHighlight/>).toJSON()
        expect(renderedComponent).toMatchSnapshot()
    })
})
