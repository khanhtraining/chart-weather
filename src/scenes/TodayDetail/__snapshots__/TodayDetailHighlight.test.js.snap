// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for Location component 1`] = `
<div
  className="todayDetailHighlight-wrapper"
>
  <div
    className="todayDetailHighlight-wrapper__icon"
  >
    <img
      alt="cloud-icon"
      src="http://openweathermap.org/img/wn/undefined@2x.png"
    />
  </div>
  <div
    className="todayDetailHighlight-wrapper__info"
  >
    <div
      className="highlight-text"
    />
    <div
      className="highlight-info"
    >
      <span
        className="highlight-temperature"
      >
        NaN
        °C
      </span>
      <span
        className="highlight-humidity"
      >
        %
      </span>
    </div>
  </div>
</div>
`;

exports[`snapshot testing snapshot for Today Detail Highlight component 1`] = `
<div
  className="todayDetailHighlight-wrapper"
>
  <div
    className="todayDetailHighlight-wrapper__icon"
  >
    <img
      alt="cloud-icon"
      src="http://openweathermap.org/img/wn/undefined@2x.png"
    />
  </div>
  <div
    className="todayDetailHighlight-wrapper__info"
  >
    <div
      className="highlight-text"
    />
    <div
      className="highlight-info"
    >
      <span
        className="highlight-temperature"
      >
        NaN
        °C
      </span>
      <span
        className="highlight-humidity"
      >
        %
      </span>
    </div>
  </div>
</div>
`;

exports[`snapshot testing snapshot for today detail highlight component 1`] = `
<div
  className="todayDetailHighlight-wrapper"
>
  <div
    className="todayDetailHighlight-wrapper__icon"
  >
    <img
      alt="cloud-icon"
      src="http://openweathermap.org/img/wn/undefined@2x.png"
    />
  </div>
  <div
    className="todayDetailHighlight-wrapper__info"
  >
    <div
      className="highlight-text"
    />
    <div
      className="highlight-info"
    >
      <span
        className="highlight-temperature"
      >
        NaN
        °C
      </span>
      <span
        className="highlight-humidity"
      >
        %
      </span>
    </div>
  </div>
</div>
`;
